kratos/tests/materials_with_subproperties.json

```diff
@@ -0,0 +1,111 @@
+{
+    "properties" : [{
+        "model_part_name" : "Inlets",
+        "properties_id"   : 1,
+        "Material"        : {
+            "constitutive_law" : {
+                "name" : "HyperElastic3DLaw",
+                "name_after_merge_rule_of_mixtures" : "RuleOfMixturesLaw",
+                "combination_factors"      : [0.4, 0.3, 0.3 ],
+                "material_rotation_angles" : [ 0,   45,  90 ],
+                "other_features..."        : [null,null,null]
+            },
+            "Variables"        : {
+                "THICKNESS"     : null,
```

Comment:
what does this do?
Together with your change in the ReadMaterialsProcess this does nothing ...

Reply:
Now I remember why ...

Reply:
"other_features..." was just an example. It may be removed...

The variables that are set to null, is to indicate that this variable will be calculated later by the composite CL in terms of the subproperties (sub-components).   Anyway, this is open to discussion...

Reply:
take care: a CL CAN NOT change the property (a propery is shared between many gauss points and hence very many CL instances!!)

Reply:
ok, I understand. So we can remove all `null` or `@auto`  variables from the properties. They could be defined at the post-processing stage.